### PR TITLE
Healthcheck and vulnerability updates

### PR DESCRIPTION
**Pull request by ives1227**
**Healthchecks and vulnerability updates.**
* * *

**JIRA Ticket**: https://jira.huit.harvard.edu/browse/LTSMAINT-646

# What does this Pull Request do?
Adds healthcheck and updates vulnerable gems.  The Ruby (JRuby) was bumped as was the Rails version was bumped which required a number of config and code changes.

# How should this be tested?

Deployed to dev/qa.  Verify that dev/qa links work here: https://wiki.harvard.edu/confluence/pages/viewpage.action?spaceKey=LibraryTechServices&title=SysDev+-+ListView.

More DRS Links can be found in Web Admin (dev/qa) by searching for Content Model = PDS Document List

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? no
- integration tests? no

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -37,18 +44,22 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
...
+  chown app /etc/ssl/openssl.cnf && \
+  chmod -R +wx /usr/local/rvm/gems/jruby-9.4.0.0
   
 
 USER app 
 
 # Set working directory
 WORKDIR /home/app/webapp
 
+USER root
```
Comment (pobocks) [inline Dockerfile:56]: This image appears to be running as root - unless that's actively desired I think the last user statement ought to be 

```dockerfile
USER app
```

File: config/initializers/load_config.rb [hunk 2]
```diff
@@ -1 +1,5 @@
-APP_CONFIG = YAML.load(ERB.new(File.read(Rails.root.join('config/config.yml'))).result)[Rails.env]
\ No newline at end of file
+begin
+  APP_CONFIG = YAML.load(ERB.new(File.read(Rails.root.join('config/config.yml'))).result, aliases: true)[Rails.env]
```
Comment (pobocks) [inline config/initializers/load_config.rb:2]: We're specifying the version of JRuby, and JRuby9.4 is equivalent to MRI 3.1, so Psych will reliably support `aliases: true` - you can take the begin/rescue out and remove the second alternative.

**Review by pobocks — CHANGES_REQUESTED** [review]
A few questions that might or might not need action.

- The container appears to be running as root, I think we want to avoid this if possible?
- Psych reliably supports `aliases: true` now so the guard around it shouldn't be necessary.

File: config/initializers/load_config.rb (hunk 2 above)
Comment (ives1227) [reply config/initializers/load_config.rb]: Fixed

File: Dockerfile (hunk 1 above)
Comment (pobocks) [reply Dockerfile:57]: Unless I'm reading this wrong, this still has `USER root` below `USER app`?  I think moving `USER app` to below line 63  would resolve this.

File: DockerfileLocal [hunk 3]
```diff
@@ -14,41 +14,45 @@ RUN mv /etc/apt/sources.list.d /etc/apt/sources.list.d.bak && \
...
 
-RUN bundle install
-
-ENTRYPOINT ["bin/localentrypoint.sh"]
-
 USER root
 
-# Expose ports
-EXPOSE 3002:3001
+RUN bundle install
```
Comment (pobocks) [inline DockerfileLocal:52]: This shouldn't afaik require root - does it (due to some passenger business) or is it a leftover?

**Review by pobocks — CHANGES_REQUESTED** [review]

File: DockerfileLocal (hunk 3 above)
Comment (ives1227) [reply DockerfileLocal:52]: Fixed the User root issue.  I think I lost the other user when I merged.

bundle install has a bunch of Jruby permission issues if I don't run it as root.  I can chmod or chown the directories but it seems to complain even when I do.  Root for that is OK.

**Review by pobocks — APPROVED** [review]

**Pull request merged**